### PR TITLE
Feature/df 1211  user changed event

### PR DESCRIPTION
**Pull request by emina-str**
other PR : https://github.com/alv-ch/alv-onlineform-service/pull/76

### CODE REVIEW

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java [hunk 1]
```diff
@@ -68,13 +69,20 @@ private UserInfo findOrCreateUserInfo(User user) {
...
             return this.userInfoRepository.saveAndFlush(userInfo1);
         } else {
-            userInfo.get().loginWithUpdate(
+            UserInfo userInfo = userInfoOptional.get();
+
+            if (userInfo.isStesUserWithUpdatedInfo(user.getFirstName(), user.getLastName(), user.getEmail())) {
+                this.applicationEventPublisher.publishEvent(new UserChangedEvent(userInfoOptional.get()));
+            }
+
+            userInfo.loginWithUpdate(
```
Comment (pavax) [inline jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java]: could the loginWithUpdate just return a boolean wheter it updated the user or not and based on that this service emits a event?

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java [hunk 2]
```diff
@@ -68,13 +69,20 @@ private UserInfo findOrCreateUserInfo(User user) {
             );
             return this.userInfoRepository.saveAndFlush(userInfo1);
         } else {
-            userInfo.get().loginWithUpdate(
+            UserInfo userInfo = userInfoOptional.get();
+
+            if (userInfo.isStesUserWithUpdatedInfo(user.getFirstName(), user.getLastName(), user.getEmail())) {
```
Comment (pavax) [inline jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java]: I'd publish all user changed event's not just the stes users

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/service/impl/messaging/UserEventGateway.java [hunk 3]
```diff
@@ -29,4 +30,10 @@ void handleUserUnregisteredEvent(UserUnregisteredEvent userUnregisteredEvent) {
         this.messageChannel.send(MessageBuilder.withPayload(userUnRegisteredEventDto).build());
     }
 
+    @EventListener
+    void handleUserChangedEvent(UserChangedEvent userChangedEvent) {
+        UserEventDto userChangedEventDto = UserEventAssembler.toUserEventDto(userChangedEvent);
```
Comment (pavax) [inline jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/service/impl/messaging/UserEventGateway.java]: id also add the 

```java
if (event.getPersonNumber() == null) {
			LOGGER.info("UserEvent for User: {} did not contain a PersonNumber -> dropped it", event.getUserInfoId());
			return;
		}
```

Comment (emina-str) [reply jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/service/impl/messaging/UserEventGateway.java]: added.

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java (hunk 2 above)
Comment (emina-str) [reply jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java]: fixed.

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java (hunk 1 above)
Comment (emina-str) [reply jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/LoginFormUserDetailsService.java]: added the return parameter, and fix the implementation

**Pull request merged**